### ISSUE TITLE
OIDC failing with Grocery Sync when creating new google project

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/couchbase-lite-ios#1621]
Google has rolled out changes with regard to webviews and oidc. I created a new google web application and started seeing these errors when testing. This could be an issue with the sample app but something we should look into. 

<img width="412" alt="screenshot 2017-02-15 14 49 14" src="https://cloud.githubusercontent.com/assets/1365736/22998944/0058860c-f38e-11e6-975f-7b0f849cf423.png">

--------
* Version:   1.4-40
* Client OS: iOS
* Server:    SG 1.4-89

**Comment by pasin** [comment]
@jamiltz how is your experiment with Safari Controller? Can we use the solution into CBL?

**Comment by pasin** [comment]
The OIDC ViewController we provide with iOS is an extra convenience feature that the other platform doesn't include. I believe the fix is to use the new Safari Controller that @jamiltz has been experimented with.

@djpongh do you think this should be fixed in 1.4?

**Comment by borrrden** [comment]
The workaround is to set the `customUserAgent` property of the web view to

> Mozilla/5.0 (iPhone; CPU iPhone OS 9_0 like Mac OS X) AppleWebKit/601.1.32 (KHTML, like Gecko) Mobile/13A4254v

**Comment by borrrden** [comment]
@pasin Safari controller will be enough if just iOS is used, but if sync gateway is meant to have multiple platforms it will require one provider for iOS and one for everything else (unless the web views of other platforms will let the user respond to custom app URLs) and we don't currently have a way to choose via Couchbase Lite client.

**Comment by jamiltz** [comment]
There's also some detail about Safari Controller in https://github.com/couchbase/couchbase-lite-ios/issues/1600 and that issue is tagged with 1.4.1.

TL;DR there's no easy way to detect when the redirect URL is reached with the Safari Controller API. I've read blogs about using a URL scheme instead which is iOS specific but couldn't get it to work. And as @borrrden pointed out, the `callback_url` in the config file would then be specific to iOS.

As for Android, Google Auth will also require the app to use [chrome custom tabs](https://developer.chrome.com/multidevice/android/customtabs) instead of a classic webview. I don't know if Chrome custom tabs have the same constraints as Safari Controller with regards to detecting when the redirect URL is reached.

**Comment by jamiltz** [comment]
Duplicate of https://github.com/couchbase/couchbase-lite-ios/issues/1600

**Comment by pasin** [comment]
From chatting with @djpongh, we will implement the user agent workaround for 1.4.
